pkg/rpctimeout/item_rpc_timeout.go

```diff
@@ -84,3 +87,49 @@ func (r *RPCTimeout) ConnectTimeout() time.Duration {
 func (r *RPCTimeout) ReadWriteTimeout() time.Duration {
 	return time.Duration(r.RPCTimeoutMS) * time.Millisecond
 }
+
+// NewContainer build Container for timeout provider.
+func NewContainer() *Container {
+	c := &Container{}
+	rtc := &rpcTimeoutConfig{
+		configs:      map[string]*RPCTimeout{},
+		globalConfig: CopyDefaultRPCTimeout().(*RPCTimeout),
+	}
+	c.config.Store(rtc)
+	return c
+}
+
+type rpcTimeoutConfig struct {
+	configs      map[string]*RPCTimeout
+	globalConfig *RPCTimeout
+}
+
+// Container is the implementation of rpcinfo.TimeoutProvider.
+// Provide the ability to dynamically configure the rpctimeout
+// config on the method hierarchy.
+type Container struct {
+	config atomic.Value
+}
+
+// NotifyPolicyChange to receive policy when it changes
+func (c *Container) NotifyPolicyChange(configs map[string]*RPCTimeout) {
+	rtc := &rpcTimeoutConfig{
+		configs:      configs,
+		globalConfig: CopyDefaultRPCTimeout().(*RPCTimeout),
+	}
+
+	// if has wildcardMethod rpctimeout config, overwrite the default one.
+	if config, ok := configs[wildcardMethod]; ok {
+		rtc.globalConfig = config
+	}
+	c.config.Store(rtc)
+}
+
+// Timeouts return the rpc timeout config by the method name of rpc info.
+func (c *Container) Timeouts(ri rpcinfo.RPCInfo) *RPCTimeout {
+	rtc := c.config.Load().(*rpcTimeoutConfig)
+	if config, ok := rtc.configs[ri.Invocation().MethodName()]; ok {
```

Comment:
Is there a concurrent read and write issue  for the config map

Reply:
For robustness, it's better we copy this map in `NotifyPolicyChange` method

Reply:
Fixed.